e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/B_PublishMultipleNodesOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,387 @@
+                    var nodeId = propertyName.Split('#').Last();
+                    var expected = expectedNodes.FirstOrDefault(n => n.EndsWith(nodeId));
+                    if (expected != null) {
+                        expectedNodes.Remove(expected);
+                    } else {
+                        unexpectedNodesThatPublish.Add(propertyName);
+                    }
+                }
+
+                expectedNodes.ForEach(n => _context.OutputHelper.WriteLine(n));
+                Assert.Empty(expectedNodes);
+
+                unexpectedNodesThatPublish.ForEach(node => _context.OutputHelper.WriteLine($"Publishing from unexpected node: {node}"));
+            }
+        }
+
+        [Fact, PriorityOrder(55)]
+        public void Test_BulkUnpublishedNodes_Expect_Success() {
+            // used if running test cases separately (during development)
+            if (string.IsNullOrWhiteSpace(_context.OpcUaEndpointId)) {
+                Test_PrepareTestDeploymentForTestCase_Expect_Success();
+                Assert.False(string.IsNullOrWhiteSpace(_context.OpcUaEndpointId));
+            }
+
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var accessToken = TestHelper.GetTokenAsync(_context, cts.Token).GetAwaiter().GetResult();
+            var client = new RestClient(_context.IIoTPlatformConfigHubConfig.BaseUrl) {
+                Timeout = TestConstants.DefaultTimeoutInMilliseconds
+            };
+
+            var request = new RestRequest(Method.POST);
+            request.AddHeader(TestConstants.HttpHeaderNames.Authorization, accessToken);
+            request.Resource = string.Format(TestConstants.APIRoutes.PublisherBulkFormat, _context.OpcUaEndpointId);
+
+            var testPlc = _context.SimulatedPublishedNodes[_context.ConsumedOpcUaNodes.First().Key];
+            _context.ConsumedOpcUaNodes.First().Value.OpcNodes = testPlc.OpcNodes.Skip(250).ToArray();
+            var body = new {
+                NodesToRemove = _context.ConsumedOpcUaNodes.First().Value.OpcNodes.Select(node => node.Id ).ToArray()
+            };
+
+            request.AddJsonBody(JsonConvert.SerializeObject(body));
+
+            var response = client.ExecuteAsync(request, cts.Token).GetAwaiter().GetResult();
+            Assert.NotNull(response);
+
+            if (!response.IsSuccessful) {
+                _output.WriteLine($"StatusCode: {response.StatusCode}");
+                _output.WriteLine($"ErrorMessage: {response.ErrorMessage}");
+                Assert.True(response.IsSuccessful, "POST /publisher/v2/publish/{endpointId}/bulk failed!");
+            }
+        }
+
+        [Fact, PriorityOrder(56)]
+        public void Test_VerifyNoDataIncomingAtIoTHub() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+            Task.Delay(90 * 1000, cts.Token).GetAwaiter().GetResult(); //wait till the publishing has stopped
```

Comment:
>90 * 1000 [](start = 23, length = 9)

magic number

Reply:
no, this is the default timeout, changed to use constant TestConstants.DefaultTimeoutInMilliseconds